grammars/css.cson

```diff
@@ -20,294 +20,1237 @@
+        'begin': '\\A(?:\\xEF\\xBB\\xBF)?(?i:(?=\\s*@charset\\b))'
+        'end': ';|(?=$)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.terminator.rule.css'
+        'name': 'meta.at-rule.charset.css'
+        'patterns': [
+          {
+            'captures':
+              '1':
+                'name': 'invalid.illegal.not-lowercase.charset.css'
+              '2':
+                'name': 'invalid.illegal.leading-whitespace.charset.css'
+              '3':
+                'name': 'invalid.illegal.no-whitespace.charset.css'
+              '4':
+                'name': 'invalid.illegal.whitespace.charset.css'
+              '5':
+                'name': 'invalid.illegal.not-double-quoted.charset.css'
+              '6':
+                'name': 'invalid.illegal.unclosed-string.charset.css'
+              '7':
+                'name': 'invalid.illegal.unexpected-characters.charset.css'
+            'match': '''(?x)        # Possible errors:
+              \\G
+              ((?!@charset)@\\w+)   # Not lowercase (@charset is case-sensitive)
+              |
+              \\G(\\s+)             # Preceding whitespace
+              |
+              (@charset\\S[^;]*)    # No whitespace after @charset
+              |
+              (?<=@charset)         # Before quoted charset name
+              (\\x20{2,}|\\t+)      # More than one space used, or a tab
+              |
+              (?<=@charset\\x20)    # Beginning of charset name
+              ([^";]+)              # Not double-quoted
+              |
+              ("[^"]+$)             # Unclosed quote
+              |
+              (?<=")                # After charset name
+              ([^;]+)               # Unexpected junk instead of semicolon
+            '''
+          }
+          {
+            'captures':
+              '1':
+                'name': 'keyword.control.at-rule.charset.css'
+              '2':
+                'name': 'punctuation.definition.keyword.css'
+            'match': '((@)charset)(?=\\s)'
+          }
+          {
+            'begin': '"'
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.definition.string.begin.css'
+            'end': '"|$'
+            'endCaptures':
+              '0':
+                'name': 'punctuation.definition.string.end.css'
+            'name': 'string.quoted.double.css'
+            'patterns': [
+              {
+                'begin': '(?:\\G|^)(?=(?:[^"])+$)'
+                'end': '$'
+                'name': 'invalid.illegal.unclosed.string.css'
+              }
+            ]
+          }
+        ]
+      }
+      {
+        # @import
+        'begin': '(?i)((@)import)(?:\\s+|$|(?=[\'"]|/\\*))'
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.import.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'end': ';|(?=})'
```

Comment:
Since @50Wliu axed nested selectors and crushed @thomaslindstrom's [hopes and dreams](https://github.com/atom/language-css/issues/51), this half of the lookahead is well-and-truly redundant, yeah.